### PR TITLE
new Rust Axum quickstarter

### LINKED ISSUE #980: Rust quickstarter
**Is your feature request related to a problem? Please describe.**
We would like to have a Rust 🦀 ODS quickstarter. See [Rust](https://www.rust-lang.org/) docs to understand more.

**Describe the solution you'd like**
A boilerplate with a minimal API example to get faster to code, with as minimal dependencies as possible and hints on what else is recommended to use, so users can later decide on their needs.

**Describe alternatives you've considered**
N/A

**Additional context**
We will provide the Rust quickstarter with [axum](https://github.com/tokio-rs/axum) web application framework.

### PR DESCRIPTION
**Pull request by gerardcl**
Adding the new ODS Rust quickstarter. Enabling teams to develop high performant and safe software with high level and zero-cost abstractions!

Closes #980 

Tasks: 
- [x] Updated documentation in `docs/modules/...` directory
- [x] Ran tests in `<quickstarter>/testdata` directory

FYI @flightlama 

@braisvq1996 please let us know if there is something missing and also if you could run a provision test on your side that would be great! let me know if I can do something regarding the testing too! 🚀

### CODE REVIEW

**Comment by braisvq1996** [comment]
I will be going into PTO so not much time to check and test the new QS. I will take a look next year :wink:

**Comment by gerardcl** [comment]
Rust 1.75.0 was released (check [here](https://blog.rust-lang.org/2023/12/28/Rust-1.75.0.html)) - so I updated the MSRV accordingly 🚀 
Happy New Year Everyone! ✨

**Comment by gerardcl** [comment]
ok we just now passed successfully all ODS tests and @braisvq1996 will also further test with the RM 🔝 
thanks @braisvq1996  for the awesome support!! 🚀 
@FLIGHTLAMA waiting for your review too! 💯

File: be-rust-axum/rust-template/src/config/settings.rs
Comment (kuebler) [inline be-rust-axum/rust-template/src/config/settings.rs:31]: Will panic if the PORT variable is assigned a string not parsable as an u16 (e.g. PORT=foo). Should we catch this and also gracefully assign 8080 in those cases?

File: be-rust-axum/rust-template/src/config/settings.rs
Comment (kuebler) [inline be-rust-axum/rust-template/src/config/settings.rs:44]: Is `DEBUG` a good default? Would rather use `INFO` or `WARN` to not have super verbose output on default?

File: be-rust-axum/rust-template/src/config/settings.rs
Comment (gerardcl) [reply be-rust-axum/rust-template/src/config/settings.rs]: fixed

File: be-rust-axum/rust-template/src/config/settings.rs
Comment (gerardcl) [reply be-rust-axum/rust-template/src/config/settings.rs:41]: fixed

**Comment by braisvq1996** [comment]
> ok we just now passed successfully all ODS tests and @braisvq1996 will also further test with the RM 🔝 thanks @braisvq1996 for the awesome support!! 🚀 @FLIGHTLAMA waiting for your review too! 💯

With the last commit now rust qs works with release manager orchestration pipeline :star:

**Review by braisvq1996 — DISMISSED** [review]
:rocket:

File: be-rust-axum/rust-template/chart/templates/_helpers.tpl
Comment (gerardcl) [inline be-rust-axum/rust-template/chart/templates/_helpers.tpl:49]: @serverhorror  this is definitely something to take note for the other helm implementations!
kudos to @braisvq1996! 🚀

**Review by FLIGHTLAMA — DISMISSED** [review]
Looks good, had only one minor question

File: Makefile
Comment (braisvq1996) [reply Makefile:60]: I forgot to update the comments on the makefile when I added nodejs 20 agent and removed nodejs12  :sweat_smile:

**Comment by metmajer** [comment]
Nice job @gerardcl and colleagues. FYI @jorge-romero @jafarre-bi. Gerard, let's speak about getting this community QS onto our Marketplace.

**Review by braisvq1996 — APPROVED** [review]

**Comment by braisvq1996** [comment]
> Nice job @gerardcl and colleagues. FYI @jorge-romero @jafarre-bi. Gerard, let's speak about getting this community QS onto our Marketplace.

As mentioned on our company chat, this quickstarter will not be a community but a fully qualified one instead.
We would still need to prepare the product page for Marketplace but the rest is the same as ussual.

**Pull request merged**
